components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -468,6 +468,27 @@ mod parser {
             _ => None,
         }
     }
+
+    /// Try to parse a `u64` into a `Time` with given type and fsp
+    pub fn parse_from_u64(
+        ctx: &mut EvalContext,
+        input: u64,
+        time_type: TimeType,
+        fsp: u8,
+    ) -> Option<Time> {
+        let aligned = match input {
+            0 => 0,
+            101..=691_231 => (input + 20_000_000) * 1_000_000,
+            700_101..=991_231 => (input + 19_000_000) * 1_000_000,
+            10_000_101..=99_991_231 => input * 1_000_000,
+            101_000_000..=691_231_235_959 => input + 20_000_000_000_000,
+            700_101_000_000..=991_231_235_959 => input + 19_000_000_000_000,
+            10_000_101_000_000..=std::u64::MAX => input,
```

Comment:
The `time_type` will be changed to `Datetime` according to TiDB.

Reply:
What about the input between 991231235959 and 10000101000000